homeassistant/components/bmw_connected_drive/__init__.py

```diff
@@ -89,22 +87,6 @@
 UNDO_UPDATE_LISTENER = "undo_update_listener"
 
 
-async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
-    """Set up the BMW Connected Drive component from configuration.yaml."""
-    hass.data.setdefault(DOMAIN, {})
-    hass.data[DOMAIN][DATA_HASS_CONFIG] = config
-
-    if DOMAIN in config:
```

Comment:
Did we log a warning somewhere that the YAML config is deprecated? We normally log a warning for two releases before removing the YAML import.

Reply:
We didn't log. However it has been included since [2021.1](https://www.home-assistant.io/blog/2021/01/06/release-20211/#integrations-now-available-to-set-up-from-the-ui) and got automatically migrated since this release.

So while it should have technically been logged, I would argue that is has been migrated for everybody using it anyhow.

Reply:
What we can do is use the `cv.removed` config validator in the `CONFIG_SCHEMA` with `raise_if_present` set to `False`, and we keep the intact schema around for two releases. This will log an error but not fail the validation.

Reply:
So we should keep this PR on hold and I'll create a new one with `cv.removed`?

Reply:
We can add that here.